backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ItemStockSammMapper.java

```diff
@@ -75,35 +76,32 @@ private ItemStockSamm listToItemStockSamm(List<? extends ItemStock> itemStocks,
                     itemStock.getNonNullCustomerOrderId(), itemStock.getNonNullSupplierOrderId(),
                     itemStock.getNonNullCustomerOrderPositionId())));
         ItemStockSamm samm = new ItemStockSamm();
-        samm.setMaterialGlobalAssetId(material.getMaterialNumberCx());
-
-        String partnerMaterialNumber = mprService.find(material, partner).getPartnerMaterialNumber();
-        String customerMatNbr = directionCharacteristic ==
-            DirectionCharacteristic.INBOUND ? material.getOwnMaterialNumber() : partnerMaterialNumber;
-        String supplierMatNbr = directionCharacteristic ==
-            DirectionCharacteristic.INBOUND ? partnerMaterialNumber : material.getOwnMaterialNumber();
-        samm.setMaterialNumberCustomer(customerMatNbr);
-        samm.setMaterialNumberSupplier(supplierMatNbr);
+
+        if (directionCharacteristic == DirectionCharacteristic.INBOUND) {
+            samm.setMaterialGlobalAssetId(mprService.find(material, partner).getPartnerCXNumber());
+        } else {
+            samm.setMaterialGlobalAssetId(material.getMaterialNumberCx());
```

Comment:
To discuss: currently we also set the mpr.partnerCXNumber, when creating mprs for customers. Wouldn't it be then more straight forward to always map from mprService? We currently have a redundant information and should work with it in one way.

Reply:
I'd say that would almost work... but not in an edge case, where two partners are effectively traders, who are each others customer and supplier at the same time ;)

Reply:
Do we want to support such an edge case? @tom-rm-meyer-ISST @eschrewe

Reply:
Well we have supported that case from the earliest day in this project :)
That's why we have, for example, allowed for the possibiliy that a Material entity can have the "product" flag and the "material" flag set to true at the same time. The same goes for the MaterialPartnerRelation Entity, where a Partner can be set as a supplier and as a customer at the same time. There is no reason to change this now, IMHO.